src/vmm/src/acpi/mod.rs

```diff
@@ -0,0 +1,282 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use acpi_tables::fadt::{FADT_F_HW_REDUCED_ACPI, FADT_F_PWR_BUTTON, FADT_F_SLP_BUTTON};
+use acpi_tables::{Dsdt, Fadt, Madt, Rsdp, Sdt, Xsdt};
+use log::{debug, error};
+use vm_allocator::AllocPolicy;
+
+use crate::acpi::x86_64::{
+    apic_addr, rsdp_addr, setup_arch_dsdt, setup_arch_fadt, setup_interrupt_controllers,
+};
+use crate::device_manager::resources::ResourceAllocator;
+use crate::vstate::memory::{GuestAddress, GuestMemoryMmap};
+use crate::Vcpu;
+
+mod x86_64;
+
+// Our (Original Equipment Manufacturer" (OEM) name. OEM is how ACPI names the manufacturer of the
+// hardware that is exposed to the OS, through ACPI tables. The OEM name is passed in every ACPI
+// table, to let the OS know that we are the owner of the table.
+const OEM_ID: [u8; 6] = *b"FIRECK";
+
+// In reality the OEM revision is per table and it defines the revision of the OEM's implementation
+// of the particular ACPI table. For our purpose, we can set it to a fixed value for all the tables
+const OEM_REVISION: u32 = 0;
+
+// This is needed for an entry in the FADT table. Populating this entry in FADT is a way to let the
+// guest know that it runs within a Firecracker microVM.
+const HYPERVISOR_VENDOR_ID: [u8; 8] = *b"FIRECKVM";
+
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+/// Error type for ACPI related operations
+pub enum AcpiError {
+    /// Could not allocate resources: {0}
+    VmAllocator(#[from] vm_allocator::Error),
+    /// ACPI tables error: {0}
+    AcpiTables(#[from] acpi_tables::AcpiError),
+}
+
+/// Helper type that holds the guest memory in which we write the tables in and a resource
+/// allocator for allocating space for the tables
+struct AcpiTableWriter<'a> {
+    mem: &'a GuestMemoryMmap,
+    resource_allocator: &'a mut ResourceAllocator,
+}
+
+impl<'a> AcpiTableWriter<'a> {
+    /// Write a table in guest memory
+    ///
+    /// This will allocate enough space inside guest memory and write the table in the allocated
+    /// buffer. It returns the address in which it wrote the table.
+    fn write_acpi_table<S>(&mut self, table: &mut S) -> Result<u64, AcpiError>
+    where
+        S: Sdt,
+    {
+        let addr = self.resource_allocator.allocate_system_memory(
+            table.len().try_into().unwrap(),
+            1,
+            AllocPolicy::FirstMatch,
+        )?;
+
+        table
+            .write_to_guest(self.mem, GuestAddress(addr))
+            .inspect_err(|err| error!("acpi: Could not write table in guest memory: {err}"))?;
+
+        debug!(
+            "acpi: Wrote table ({} bytes) at address: {:#010x}",
+            table.len(),
+            addr
+        );
+
+        Ok(addr)
+    }
+
+    /// Build the DSDT table for the guest
+    fn build_dsdt(&mut self) -> Result<u64, AcpiError> {
+        debug!("acpi: Building DSDT table");
```

Comment:
Do we really need `debug` logs? They only mention the function they are called from, but don't provide any additional info. Maybe lower them to `trace` level if you think they are still valuable?

Reply:
dropped the debug macros that just print the function we're in. I don't believe it's even worth having them as tracepoints. These functions are very very small.